src/accelerate/data_loader.py

```diff
@@ -1164,12 +1180,15 @@ class SkipDataLoader(DataLoaderAdapter):
     def __init__(self, dataset, skip_batches=0, use_stateful_dataloader=False, **kwargs):
         super().__init__(dataset, use_stateful_dataloader=use_stateful_dataloader, **kwargs)
         self.skip_batches = skip_batches
+        self.gradient_state = GradientState()
```

Comment:
It would be helpful if `DataLoaderStateMixin` docstring mentioned that `self.gradient_state` is required.